File: CS2242_SMART_WHEELCHAIR/src/system/CareTaker.java
package system;

public class CareTaker extends Person{
	private String role;
	private String userName;//login credentials
	
	public CareTaker(String name, int nic, String address, int age, String gender, CareHouse relatedCareHouse,ContactDetail contactDetail,String role,String userName) {
		super(name, nic, address, age, gender, relatedCareHouse, contactDetail);
		this.role =role;
		this.userName =userName;
	}

	protected String getRole() {
		return role;
	}

	protected String getUserName() {
		return userName;
	}
	
	protected void changeUserName(String userName) {
		this.userName = userName;
	}
	
}

File: Thilina/C++ Classes/care_taker.cpp
#include"care_taker.h"

CareTaker::CareTaker(const std::string& name, const std::string& nic, const std::string& address, 
                    const int& age, Gender gender, CareHouse* related_care_house, const std::string& roll)
:Person(name, nic,  address, age, gender, related_care_house), roll(roll){

}

const std::string& CareTaker::get_roll() const{
    return roll;
}